ext/rugged/rugged_submodule.c

```diff
@@ -554,7 +554,7 @@ static VALUE rb_git_submodule_url(VALUE self)
 
 	url = git_submodule_url(submodule);
 
-	return url ? rb_str_new_utf8(url) : Qnil;
+	return url ? rb_str_new2(url) : Qnil;
```

Comment:
I think URLs should be US-ASCII.

Reply:
Pretty sure.  I think IDNs use [Punycode](https://en.wikipedia.org/wiki/Punycode).

Also:

<img width="608" alt="qmk_firmware fish users aaron git qmk_firmware -fish 247x51 2019-02-15 08-11-20" src="https://user-images.githubusercontent.com/3124/52868903-51ddef80-30f9-11e9-949b-8810e81b28f1.png">

Reply:
(that said, I have no idea if git actually enforces it)

Reply:
I think git would pass whatever garbage over to libcurl and have it figure it out.